### PR TITLE
Get back all neighbors for a stone

### PR DESCRIPTION
**Pull request by Chroq**
## Description

In this PR, I get back all neighbors to build the graph (first commit). After that, we build a graph based on the neighbors.

Trello : [https://trello.com/c/IOebRRnb/1-1as-norbert-given-a-game-state-i-want-to-know-if-i-have-a-chance-of-winning-or-not-on-the-next-turn](url)

`make test`

### CODE REVIEW

File: src/hex/neighbors_test.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+		{0, 1, 1, 0, 2},
+		{0, 1, 2, 2, 2},
+		{0, 1, 0, 0, 2},
+		{1, 1, 1, 0, 2},
+		{1, 0, 0, 0, 2},
+	}
+
+	stones := getStonesFromMatrix(matrix)
+
+	stone := *NewStone(1, 1, 1, Player1)
```
Comment (JulienMattiussi) [inline src/hex/neighbors_test.go]: I think it's id 7 and not 1 for this stone

Comment (JulienMattiussi) [inline src/hex/neighbors_test.go]: You should add a test for an edge stone to valid getNeighborsForStone behaviour

File: src/hex/neighbors.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+func NewNeighbor(stone Stone, distance int) *Neighbor {
+	return &Neighbor{stone, distance}
+}
+
+func getNeighborsForStone(stones []Stone, stone Stone) []Neighbor {
+	directions := getDirections()
+	directionCount := len(directions)
+
+	neighbors := make([]Neighbor, directionCount)
```
Comment (JulienMattiussi) [inline src/hex/neighbors.go]: no need of neighborsCount IMO
You can start with empty slice and append until having the full size
https://tour.golang.org/moretypes/15

File: src/hex/neighbors_test.go (hunk 1 above)
Comment (Chroq) [reply src/hex/neighbors_test.go]: Changed ! TY

Comment (Chroq) [reply src/hex/neighbors_test.go]: Added ! Ty

File: src/hex/neighbors.go (hunk 2 above)
Comment (Chroq) [reply src/hex/neighbors.go]: Yes, removed ! Ty
